WordPress/Sniffs/Theme/NoTitleTagSniff.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of the <title> tag, unless it is within a <svg> tag.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoTitleTagSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Property to keep track of whether a <svg> open tag has been encountered.
+	 *
+	 * @var bool
+	 */
+	private $in_svg = false;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_INLINE_HTML, // Finds inline html.
+			T_CONSTANT_ENCAPSED_STRING, // Finds the title tag in PHP.
+		 );
+	} // end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token in
+	 *                                        the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens  = $phpcsFile->getTokens();
+		$content = $tokens[ $stackPtr ]['content'];
+
+		// No need to check an empty string.
+		if ( '' === trim( $content ) ) {
+			return;
+		}
+
+		// Are we in a <svg> tag ?
+		if ( true === $this->in_svg ) {
+			if ( false === strpos( $content, '</svg>' ) ) {
+				// Rest in_svg at the end of the file.
+				if ( ( $phpcsFile->numTokens - 1 ) === $stackPtr ) {
+					$this->in_svg = false;
+				}
+				return;
+			} else {
+				// Make sure we check any content on this line after the closing svg tag.
+				$this->in_svg = false;
+				$content      = trim( substr( $content, ( strpos( $content, '</svg>' ) + 6 ) ) );
+			}
+		}
+
+		/*
+		 * We're not in svg, so check if it there's a <svg> open tag on this line.
+		 * PHP 5.2 creates a seperate token for `<s` we need to check that the first two letters in the next token are vg.
+		*/
+		$next_token = $phpcsFile->findNext( array( T_INLINE_HTML ), ( $stackPtr + 1 ) );
```

Comment:
Using the PHPCS native function is absolutely the right way to go. Good improvement.

However, that leaves two things:
* The second part of the check only needs to be done when the first token was T_INLINE_HTML, now it could accidentally match code like: `echo '<s'; ?> vga`. (first part is T_CONSTANT_ENCAPSED_STRING, second part T_INLINE_HTML)
* Second of all: the native function call might return false, so `$next_token` might be `false`, so that needs to be checked for, otherwise you will be causing `undefined index` notices.